production/helm/loki/templates/_helpers.tpl

```diff
@@ -322,3 +329,33 @@ Create the service endpoint including port for MinIO.
 {{ include "loki.name" . }}-memberlist
 {{- end -}}
 
+{{/* Determine the public host for the Loki cluster */}}
+{{- define "loki.host" -}}
+{{- $isSingleBinary := eq (include "loki.deployment.isSingleBinary" .) "true" -}}
+{{- $url := printf "%s.%s.svc.%s" (include "loki.gatewayFullname" .) .Release.Namespace .Values.global.clusterDomain }}
+{{- if $isSingleBinary  }}
+  {{- $url = printf "%s.%s.svc.%s:3100" (include "loki.singleBinaryFullname" .) .Release.Namespace .Values.global.clusterDomain }}
```

Comment:
so I think the reasoning here is why would someone be using a gateway when running monolithic mode, since monolithic assumes a single instance (filesystem storage cannot support > 1 replica)? Maybe we should be more explicit about that requirement though, both in the check here and a validation file?

Reply:
hmm we could be more explicit definetely, but what you said is fair enough. Resolved!